### ISSUE TITLE
Search field is hidden when browser window is too small

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#4093]
**[Submitted to the original trac issue database at 6.48pm, Saturday, 19th November 2011]**

When the stylesheet for small screens is loaded (i.e. the browser window is smaller than 642px), the search box is hidden. I consider the search box to be an essential part of the website and I don't understand why it isn't shown e.g. next to the OpenStreetMap logo at the top of the page (at the moment there is a unused white space).

**Comment by tomhughes** [comment]
**Author: Richard**
**[Added to the original trac issue at 7.23pm, Saturday, 19th November 2011]**

This is a known issue.

But come to the next OSM hack weekend at http://wiki.openstreetmap.org/wiki/London/London_Hack_Weekend_Nov_2011 and you can help be part of the solution rather than just repeating the problem!

**Comment by tomhughes** [comment]
**Author: TomH**
**[Added to the original trac issue at 7.34pm, Saturday, 19th November 2011]**

I think you'll find the change is actually at 480px not 642px.

This is not something that is likely to be solvable with a simple stylesheet change though - finding somewhere for the search box (while hard) is actually the easy bit. The hard bit will be managing the search results given that it will probably need completely different handling to the search results for large screens which is something we have managed to avoid so far.

Basically what I'm saying is that this is a major project that somebody would have to want to take on.

**Comment by tomhughes** [comment]
**Author: gulp21**
**[Added to the original trac issue at 9.45pm, Saturday, 19th November 2011]**

The search bar with the search results could be shown on top of the map (covering the whole map). When the user clicks on a search result, the sidebar would be closed, so that the map is visible again. This idea might not be perfect, but it is better than the current situation, and I think implementing it doesn't require so much work.

**Comment by tomhughes** [comment]
**Author: TomH**
**[Added to the original trac issue at 12.13am, Sunday, 20th November 2011]**

Yes I'm well aware of what *could* be done. The point is that there is a large chunk of code to make it work the way it does and it would need another large chunk to make it work in another way on a mobile client.

So, as I said, this would be a large project for somebody.

**Comment by tomhughes** [comment]
**Author: TomH**
**[Added to the original trac issue at 2.32pm, Saturday, 30th November 2013]**

Search is now available on small screen devices.
